@here/harp-mapview/test/PickingRaycasterTest.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// tslint:disable:only-arrow-functions
+//    Mocha discourages using arrow functions, see https://mochajs.org/#arrow-functions
+// tslint:disable:no-unused-expression
+//    Chai uses properties instead of functions for some expect checks.
+
+import { GeometryKind } from "@here/harp-datasource-protocol";
+import { expect } from "chai";
+import * as sinon from "sinon";
+import * as THREE from "three";
+import { MapObjectAdapter } from "../lib/MapObjectAdapter";
+import { PickingRaycaster } from "../lib/PickingRaycaster";
+
+function createFakeObject<T extends THREE.Object3D>(type: new () => T): T {
+    const object = new type();
+    sinon.stub(object, "raycast").callsFake((_, intersects) => {
+        intersects.push({ distance: 0, point: new THREE.Vector3(), object });
```

Comment:
LGTM, just please add a test case for the recursive code block

Reply:
done